azurerm/internal/services/appconfiguration/app_configuration_resource.go

```diff
@@ -199,6 +224,11 @@ func resourceArmAppConfigurationCreate(d *schema.ResourceData, meta interface{})
 		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
 
+	if _, ok := d.GetOk("identity"); ok {
+		appConfigIdentity := expandAppConfigurationIdentity(d.Get("identity").([]interface{}))
+		parameters.Identity = appConfigIdentity
+	}
```

Comment:
thinking about it, since the expand func below returns `None` if no elements are present, we can remove this if statement here